src/transformers/modeling_tf_t5.py

```diff
@@ -355,16 +355,49 @@ def call(
         return outputs  # hidden-states, (self-attention weights), (self-attention position bias), (cross-attention weights), (cross-attention position bias)
 
 
+class _NoLayerEmbedTokens(object):
+    """
+     this class wraps a the TFSharedEmbeddingTokens layer into a python 'no-keras-layer'
+     class to avoid problem with weight restoring. Also it makes sure that the layer is
+     called from the correct scope to avoid problem with saving/storing the correct weights
+    """
+
+    def __init__(self, layer, abs_scope_name=None):
+        self._layer = layer
+        self._abs_scope_name = abs_scope_name
+
+    def call(self, inputs, mode="embedding"):
+        if self._abs_scope_name is None:
+            return self._layer.call(inputs, mode)
+
+        # if an abs scope name is given to the embedding variable, call variable from absolute scope
+        with tf.compat.v1.variable_scope(self._abs_scope_name, auxiliary_name_scope=False) as abs_scope_name:
+            with tf.name_scope(abs_scope_name.original_name_scope):
+                return self._layer.call(inputs, mode)
```

Comment:
Thanks a lot @craffel for the dependency injection hint for the TF scope. Implemented it according to the second example on https://www.tensorflow.org/api_docs/python/tf/compat/v1/variable_scope. 

PT & TF conversion now works without errors. 

@mfuntowicz @thomwolf @craffel 
Is the behavior of this Wrapper class ok for you?